src/compositemark/boxplot.ts

```diff
@@ -289,11 +290,13 @@ export function normalizeBoxPlot(
   }
 
   if (filteredLayersMixins) {
+    filteredLayersMixins.transform.unshift(...preJoinAggregate);
```

Comment:
Instead of adding duplicates of all of these transform to `filteredLayersMixins` 

Can we restructure this such that we only do timeUnit/ binning computation only once?

Reply:
Do you mean putting the transform outside layer or combining the filtered layer and the boxplot together?

Reply:
Yes, that should be possible?

Otherwise, we're always running the computation multiple times, which seems suboptimal.